src/deploy.js

```diff
@@ -93,14 +77,46 @@ const deploy = async (opts) => {
     console.error(
       'error: and contact your favourite Graasp engineer if you keep running into trouble',
     );
+    return false;
   }
+  return true;
+};
+
+const deploy = async (opts) => {
+  const { tag, env, build } = opts;
+
+  // validate command options
+  if (!validateTag(tag) || !validateEnv(env) || !validateBuild(build)) {
+    console.error('Abort...');
+    return false;
+  }
+
+  // set environment
+  dotenv.config({ path: env });
+
+  // validate environment variables
+  if (
+    !validateAppVariables(process.env) ||
+    !validateAwsCredentialsVariables(process.env)
+  ) {
+    return false;
+  }
+
+  const {
+    REACT_APP_GRAASP_DEVELOPER_ID,
+    REACT_APP_GRAASP_APP_ID,
+    REACT_APP_HOST,
+    REACT_APP_VERSION,
+    BUCKET,
+    DISTRIBUTION,
+  } = process.env;
 
   console.log(
     `info: publishing app ${REACT_APP_GRAASP_APP_ID} version ${REACT_APP_VERSION}`,
   );
 
   // configure the deployment
-  AWS.config.getCredentials((err) => {
+  aws.config.getCredentials((err) => {
```

Comment:
Even if you guarantee that the correct variables are present, don't you need to wait for the setup to be done?

Reply:
Indeed the execution relies currently on a fast execution of this async method. A next PR must handle this problematic and properly chain the execution.